### PR TITLE
do not adjust beatloop_size when pressing beatlooproll_X_activate

### PR DESCRIPTION
**Pull request by Be-ing**
Fixing https://bugs.launchpad.net/mixxx/+bug/1721881

### CODE REVIEW

File: src/engine/loopingcontrol.cpp [hunk 1]
```diff
@@ -910,8 +913,7 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint, bool enable
     }
 
     int samples = m_pTrackSamples->get();
-    if (!m_pTrack || samples == 0
-            || !m_pBeats) {
+    if ((!m_pTrack || samples == 0 || !m_pBeats) && adjustBeatloopSizeCO) {
         clearActiveBeatLoop();
```
Comment (daschuer) [inline src/engine/loopingcontrol.cpp]: Is it correct to skip this as well?

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/engine/loopingcontrol.cpp (hunk 1 above)
Comment (Be-ing) [reply src/engine/loopingcontrol.cpp]: Good point, no it is not. I'll just make the following line conditional for adjustBeatloopSizeCO.

File: src/engine/loopingcontrol.cpp [hunk 2]
```diff
@@ -832,7 +832,7 @@ void LoopingControl::slotBeatLoopActivateRoll(BeatLoopingControl* pBeatLoopContr
 
     // Disregard existing loops.
     m_pSlipEnabled->set(1);
-    slotBeatLoop(pBeatLoopControl->getSize(), false, true);
+    slotBeatLoop(pBeatLoopControl->getSize(), false, true, false);
```
[the remark this answers is not in the document]
Comment (Be-ing) [reply src/engine/loopingcontrol.cpp]: The beatlooproll_X_activate buttons are momentary use whereas beatloop_size is a more enduring value. @ronso0 reported it was unexpected and annoying that beatlooproll_X_activate changed beatloop_size and I think that makes sense.

**Comment by ronso0** [comment]
Ready to test?

**Comment by Be-ing** [comment]
Yes.

**Comment by ronso0** [comment]
Just tested this:
1. it's not changing beatlooop_size in the spinbox  :thumbsup:
2. 1/16 beatloop_roll only worked in 50% of tests, else it would just keep playing normally with an active loop in the waveforms  :thumbsdown:
3. when using a rolling beatloop it would highlight the loop button related to loop size in the spinbox, not the one of actual beatloop_roll size  :confused: (Tango with beatloop grid open)
4. using beatloop_roll would leave it's loop_in/_out markers, so reloop would activate that rolling beatloop

3. would be nice-to-have
4. probably won't be solved here, this would require handling multiple loops

**Comment by ronso0** [comment]
> 1/16 beatloop_roll only worked in 50% of tests, else it would just keep playing normally with an active loop in the waveforms

sorry, this also happens in master

**Comment by Be-ing** [comment]
Okay, then this is ready to merge.

> when using a rolling beatloop it would highlight the loop button related to loop size in the spinbox, not the one of actual beatloop_roll size 😕 (Tango with beatloop grid open)

That would require adding beatlooproll_X_enabled Controls separate from beatloop_X_enabled, which is beyond the scope of this little bugfix PR.

**Comment by daschuer** [comment]
>> 1/16 beatloop_roll only worked in 50% of tests, else it would just keep playing normally with an active loop in the waveforms

> sorry, this also happens in master

Ups, that is probably related to my work here: https://github.com/mixxxdj/mixxx/pull/1367

**Comment by daschuer** [comment]
> when using a rolling beatloop it would highlight the loop button related to loop size in the spinbox, not the one of actual beatloop_roll size :confused: (Tango with beatloop grid open)

This is IMHO blocking the merge, because we have beatloop roll on the right click of these buttons.

**Comment by daschuer** [comment]
Let's move this from the 2.1 milestone in favour of a consistent temporary rolling loop solution.
That is the requested feature and hacking around this  may create other issues.

**Comment by ronso0** [comment]
this happened with this PR only, so No there's no bug report.

**Comment by Holzhaus** [comment]
Hey, #1367 has been merged in the meantime. What's the status of this? Can you resolve the merge conflicts?

**Comment by Be-ing** [comment]
@Holzhaus I have no plans to work on this soon. If you want to finish it, please go ahead.

**Comment by Holzhaus** [comment]
Is there anything else to be done except fixing the merge conflicts?

**Pull request closed without merging**